source/manual/conventions-in-rails-applications.html.md

```diff
@@ -0,0 +1,301 @@
+
+GOV.UK have published a number of gems that help achieve common needs across
+Ruby / Rails applications:
+
+- [gds-api-adapters][] - provides common patterns to communicate between GOV.UK
+  applications
+- [gds-sso][] - integration with [signon][] for user and application
+  authentication
+- [govuk_app_config][] - provides common configuration for applications such as
+  logging, error handling and metric reporting
+- [govuk_message_queue_consumer][] - provides a consistent means to work with
+  the the RabbitMQ queue that the [Publishing API][] broadcasts to
+- [govuk_publishing_components][] - provides a framework for embedding common
+  GOV.UK interface components and creating your own application specific ones
+- [govuk_sidekiq][] - Provides common configuration for using sidekiq with
+  GOV.UK infrastructure
+- [govuk_schemas][] - Provides helper utilities for interfacing with
+  [govuk-content-schemas][]
+- [govuk_test][] - Provides configuration and dependencies for headless browser
+  testing
+- [plek][] - Utility tool to access determine base URLs for GOV.UK applications
+  for the current environment
+- [rubocop-govuk][] - Provides GOV.UK linting rules for Ruby, Rails, RSpec and
+  Rake
+- [scss-lint-govuk][] - Provides linting for SCSS files according to GDS
+  conventions
+- [slimmer][] - Provides consistent templating for apps that serve content on
+  the www.gov.uk host
+
+Using these help maintain consistency across the programme and enables common
+iterations to be pushed out. It is encouraged that gems are iterated or issues
+filed ([example][issue-example]) when apps need to vary from conventions
+introduced by these gems.
+
+[gds-api-adapters]: https://github.com/alphagov/gds-api-adapters
+[gds-sso]: https://github.com/alphagov/gds-sso
+[signon]: https://github.com/alphagov/signon
+[govuk_app_config]: https://github.com/alphagov/govuk_app_config
+[govuk_message_queue_consumer]: https://github.com/alphagov/govuk_message_queue_consumer
+[Publishing API]: https://github.com/alphagov/publishing-api
+[govuk_publishing_components]: https://github.com/alphagov/govuk_publishing_components
+[govuk_sidekiq]: https://github.com/alphagov/govuk_sidekiq
+[govuk_schemas]: https://github.com/alphagov/govuk_schemas
+[govuk-content-schemas]: https://github.com/alphagov/govuk-content-schemas
+[govuk_test]: https://github.com/alphagov/govuk_test
+[plek]: https://github.com/alphagov/plek
+[rubocop-govuk]: https://github.com/alphagov/rubocop-govuk
+[scss-lint-govuk]: https://github.com/alphagov/scss-lint-govuk
+[slimmer]: https://github.com/alphagov/slimmer
+[issue-example]: https://github.com/alphagov/govuk_app_config/issues/121
+
+## Data storage
+
+For non-specialist database needs you should use
+[PostgreSQL](https://www.postgresql.org/) with [ActiveRecord][]. The
+[db/seeds.rb][] can be used to populate the database for development and test
+environments.
+
+For [key-value datastore][] access (such as queues or distributed caches)
```

Comment:
When should we be using RabbitMQ as a queue?

Reply:
Thanks for taking a look @theseanything 😄 

RabbitMQ is used for inter-app communication whereas Redis is only used on a single app basis. I'm not too sure RabbitMQ usage is a convention since I think we only have the one Publishing API queue that is used for communicating between apps (unless I'm missing something). Though I can see how clarifying those different types of queue usage would be useful.

Reply:
Ah yup -  inter-app communication is a good property distinguish it with!

Reply:
I'll add some text in before this is merged.

Reply:
I've added a quick bit of info to this in 5ef55c07